src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -113,9 +116,9 @@ private static string UrlEncodeSegment(byte[] segment)
             return Base64UrlEncoder.Encode(segment);
         }
 
-        private static string EncodeHeaderToJson(IClientAssertionCertificate credential)
+        private static string EncodeHeaderToJson(IClientAssertionCertificate credential, bool sendX5C)
         {
-            JWTHeaderWithCertificate header = new JWTHeaderWithCertificate(credential);
+            JWTHeaderWithCertificate header = new JWTHeaderWithCertificate(credential, sendX5C);
```

Comment:
You don't really need the local variable header, you could write:
return JsonHelper.EncodeToJson(new JWTHeaderWithCertificate(credential, sendX5C));